src/Text/Pandoc/Writers/Markua.hs

```diff
@@ -0,0 +1,501 @@
+    return $ cr <> literal txt <> cr 
+
+blockToMarkua opts (BlockQuote blks) = do
+    contents <- blockListToMarkua opts blks
+    let quoteHead = "> "
+    return $ prefixed quoteHead (nestle contents) <> blankline
+
+blockToMarkua opts ((OrderedList (start,style,delim) items)) = do
+    let listAttributes = (start,style,delim)
+    let markers  = orderedListMarkers listAttributes
+    contents <- zipWithM (orderedListItemToMarkua opts) markers items
+    return $ cr <> mconcat contents <> blankline
+
+blockToMarkua opts (BulletList items) = do
+    contents <- mapM (bulletListItemToMarkua opts) items
+    return $ cr <> mconcat contents <> blankline
+    
+blockToMarkua opts (DefinitionList items) = do
+    contents <- mapM (definitionListItemToMarkua opts) items
+    return $ cr <> mconcat contents <> blankline
+
+-- every header gets automatically an attribute list
+blockToMarkua opts (Header level attr inlines) = do
+    contents <- inlineListToMarkua opts inlines
+    let attributes = attrToMarkua attr
+    let eqs = literal $ T.replicate level "#"
+    return $ blankline <> attributes <> cr 
+            <> eqs <> space <> contents <> blankline
+
+blockToMarkua _ HorizontalRule = return $ blankline <> "* * *" <> blankline
+
+-- div is also used to compute the bibliography
+-- if --citeproc and --bibliography is set (div with refs id is generated),
+-- then a bibliography header level 1 with a attr list above is inserted
+blockToMarkua opts (Div attr blks) = do
+    contents <- blockListToMarkua opts blks
+    let attrId = getAttrId attr
+    let attrIdPrefix = T.take 4 attrId
+    let citeId = if attrIdPrefix == "ref-"
+                 then T.drop 4 attrId
+                 else T.empty
+    let headRef = literal (if not $ T.null citeId
+                           then T.cons '[' (T.append citeId "] ")
+                           else T.empty)
+    -- attr only used for bibliography
+    let attributes = attrToMarkua (citeId,[],[])
+    let biblioHeader = literal (if attrId == "refs"
+                                then "{id: bibliography}\n#Bibliography"
+                                else T.empty)
+    -- to make this block more general,
+    -- no div tags are wrapped around the content, for this reason also no
+    -- attr list is added in the general case
+    return $ blankline <> biblioHeader <> blankline <> attributes
+                       <> cr <> headRef <> nestle contents <> blankline
+
+blockToMarkua opts (Table _ blkCapt specs thead tbody tfoot) = do
+    let (caption, aligns, widths, headers, rows) = toLegacyTable
+                                                   blkCapt
+                                                   specs
+                                                   thead 
+                                                   tbody 
+                                                   tfoot
+    let numcols = maximum (length aligns :
```

Comment:
Checking all these things is good, defensive programming. I believe that `toLegacyTable` guarantees all lengths to be equal, so the code below could be simplified.

The use of `maximum` is ok here, as the list can never be empty. However, we generally try to avoid partial functions. How about `foldr (max . length) 0 [...]`?